backend/tracim_backend/lib/webdav/resources.py

```diff
@@ -68,143 +70,224 @@ def wrapper(self: "_DAVResource", *arg, **kwarg) -> typing.Callable:
     return decorator
 
 
-class RootResource(DAVCollection):
+def get_workspace_resource(
+    path: str,
+    environ: typing.Dict,
+    workspace: Workspace,
+    label: str,
+    tracim_context: "WebdavTracimContext",
+) -> "WorkspaceResource":
+    return WorkspaceResource(
+        path=path, environ=environ, workspace=workspace, tracim_context=tracim_context, label=label,
+    )
+
+
+def get_content_resource(
+    content: Content,
+    path: str,
+    environ: typing.Dict,
+    workspace: Workspace,
+    tracim_context: "WebdavTracimContext",
+) -> _DAVResource:
     """
-    RootResource ressource that represents tracim's home, which contains all workspaces
+    Helper to get correct content webdav resource according to content type
     """
+    if content.type == content_type_list.Folder.slug:
+        return FolderResource(
+            path=path,
+            environ=environ,
+            workspace=workspace,
+            content=content,
+            tracim_context=tracim_context,
+        )
+    elif content.type == content_type_list.File.slug:
+        return FileResource(
+            path=path, environ=environ, content=content, tracim_context=tracim_context,
+        )
+    else:
+        return OtherFileResource(
+            path=path, environ=environ, content=content, tracim_context=tracim_context
+        )
 
-    def __init__(self, path: str, environ: dict, tracim_context: "WebdavTracimContext"):
-        super(RootResource, self).__init__(path, environ)
+
+class WebdavContainer(ABC):
+    """
+    A Webdav Container should implement theses methods.
+    """
+
+    @abstractmethod
+    def createEmptyResource(self, file_name: str):
+        pass
+
+    @abstractmethod
+    def createCollection(self, label: str) -> "FolderResource":
+        pass
+
+    @abstractmethod
+    def getMemberNames(self) -> [str]:
+        pass
+
+    @abstractmethod
+    def getMember(self, label: str) -> _DAVResource:
+        pass
+
+    @abstractmethod
+    def getMemberList(self) -> [_DAVResource]:
+        pass
+
+
+class WorkspaceOnlyContainer(WebdavContainer):
+    """
+    Container that can get children workspace
+    """
+
+    def __init__(
+        self,
+        path: str,
+        environ: dict,
+        label: str,
+        workspace: typing.Optional[Workspace],
+        provider: "TracimDavProvider",
+        tracim_context: "WebdavTracimContext",
+        list_orphan_workspaces: bool = False,
+    ) -> None:
```

Comment:
```suggestion
    ):
```
A constructor should never return anything else than None anyway.

Reply:
As far as I remember mypy doesn't do type checking in functions/methods which do not have a completely typed signature. @raphj Do you know if there is a special case for `__init__` that would make mypy work without the `-> None` typing?

Reply:
Oh, I didn't know and you are right:

On this code:

```python
class C:
        def __init(self) -> None:
                a: int = 2
                print('a' + a)
```
Mypy says:

```
test.py:4: error: Unsupported operand types for + ("str" and "int")
Found 1 error in 1 file (checked 1 source file)
```

and on this one:

```python
class C:
        def __init(self):
                a: int = 2
                print('a' + a)
```

Mypy says: 

```
Success: no issues found in 1 source file
```

So yeah, maybe keep `-> None`. Typescript complains on this, hence my initial remark.